### ISSUE TITLE
file module: check mode fails for non-existent links

### ISSUE THREAD

**Issue by muffl0n** [ansible/ansible#6182]
##### Issue Type:

Bug Report
##### Ansible Version:

ansible 1.4.5
##### Environment:

Gentoo Base System release 2.2
##### Summary:

Using check mode fails if the used playbook contains a task to create a link that doesn't exist yet.
##### Steps To Reproduce:

inventory file 'development':

```
localhost
```

playbook file 'development-link.yml

```
- hosts: localhost
  tasks:
    - file: src=/tmp dest=/tmp/link owner=root group=root state=link
```

call to execute playbook in check mode

```
ansible-playbook -i development development-link.yml -C
```
##### Expected Results:

ansible-playbook should show what would have been changed
##### Actual Results:

```
$ ansible-playbook -i development development-link.yml -C

PLAY [localhost] ************************************************************** 

GATHERING FACTS *************************************************************** 
ok: [localhost]

TASK: [file src=/tmp dest=/tmp/link owner=root group=root state=link] ********* 
fatal: [localhost] => failed to parse: Traceback (most recent call last):
  File "/root/.ansible/tmp/ansible-1393408630.34-275498413659402/file", line 1378, in <module>
    main()
  File "/root/.ansible/tmp/ansible-1393408630.34-275498413659402/file", line 327, in main
    changed = module.set_file_attributes_if_different(file_args, changed)
  File "/root/.ansible/tmp/ansible-1393408630.34-275498413659402/file", line 813, in set_file_attributes_if_different
    file_args['path'], file_args['owner'], changed
  File "/root/.ansible/tmp/ansible-1393408630.34-275498413659402/file", line 726, in set_owner_if_different
    orig_uid, orig_gid = self.user_and_group(path)
  File "/root/.ansible/tmp/ansible-1393408630.34-275498413659402/file", line 682, in user_and_group
    st = os.lstat(filename)
OSError: [Errno 2] No such file or directory: '/tmp/link'


FATAL: all hosts have already failed -- aborting

PLAY RECAP ******************************************************************** 
           to retry, use: --limit @/home/svs/development-link.retry

localhost                  : ok=1    changed=0    unreachable=1    failed=0
```

Even though the link does not exist ansible tries to do a os.lstat(filename).
Maybe in https://github.com/ansible/ansible/blob/devel/library/files/file#L331 the method "set_file_attributes_if_different" should not be called when in check mode?

**Comment by mpdehaan** [comment]
I would agree check mode should not do anything here.

**Comment by bcoca** [comment]
this works in current devel, probably fixed by d5fa690a which was included in 1.6.0 as far as i can tell

**Comment by muffl0n** [comment]
Tested with 1.6.8. Seems to work. Thanks!
